src/Collector/RegexCollector.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Qossmic\Deptrac\Collector;
+
+abstract class RegexCollector
+{
+    /**
+     * @param array<string, string|array> $configuration
+     */
+    abstract protected function getPattern(array $configuration): string;
+
+    /**
+     * @param array<string, string|array> $configuration
+     */
+    protected function getValidatedPattern(array $configuration): string
+    {
+        $pattern = $this->getPattern($configuration);
+        set_error_handler(static function () {}, E_WARNING);
```

Comment:
Why not just silence the warning with `@preg_match`?

Reply:
Because the silence operator will silence more than just the warning generated from preg_match. It will silence everything, including fatal errors. There is IMHO nothing more demoralizing than a script stopping for no apparent reason and no error raised.

Reply:
Do you have a concrete error message in mind that we would need here? In our case the script will not just stop for no apparent reason, we throw an exception telling users their regex is invalid. I'd much rather use a simple `@` here and then improve the exception message for edge cases if people report issues. As it's likely the default error message will not help them either.

Reply:
```php
        $isRegularExpression = false !== preg_match($pattern, '');
```
can generate a whole host of errors for a multitude of reasons. If you use `@` you silence them all, including, but not limited to, fatal errors. I do not wish to silence them all. All I am looking for is to silence warning and warning only. Therefore `set_error_handler(static function () {}, E_WARNING);` is IMHO more suitable there, since I with for all other error to be raised and not handled.

Reply:
Yes, I understand the what, but I think we disagree on the why. We will get an error, but it will likely not be helpful to users anyway. To me the exception message is enough feedback (for now) and then, when we get issues, we can improve it.

For a library I would prefer your approach, because we might accidentally hide details that can be fixed upstream, but in our case - no matter what error happens - all the users can do is fix their invalid regex in the config. I don't think we lose any relevant details by silencing. I think having a single message "Invalid regex" that can be traced back to our exception is more helpful than mixing styles (exception for warning, but original error message for fatals).

Reply:
I have made my case, I do not really care either way, so I changed it.